test/client/fixtures/core/request-barrier-test.js

```diff
@@ -0,0 +1,466 @@
+                                ok(secondRequestCompleted);
+                            });
+
+                        fetch('/xhr-test/200')
+                            .then(function (response) {
+                                return response.text();
+                            })
+                            .then(function () {
+                                secondRequestCompleted = true;
+                            });
+                    }, 100);
+                });
+
+                asyncTest('barrier - creating new iframe without src (B236650)', function () {
+                    var $iframe           = null;
+                    var windowErrorRaised = false;
+
+                    window.onerror = function () {
+                        windowErrorRaised = true;
+                    };
+
+                    var action = function (callback) {
+                        if ($iframe)
+                            $iframe.remove();
+
+                        window.setTimeout(function () {
+                            $iframe = $('<iframe id="test_unique_id_ydnvbfq2">').attr('src', 'about:blank').appendTo('body');
+                        }, 0);
+
+                        callback();
+                    };
+
+                    var requestBarrier = new RequestBarrier();
+
+                    action.call(window, function () {
+                        requestBarrier
+                            .wait()
+                            .then(function () {
+                                ok(!windowErrorRaised);
+                            });
+                    });
+
+                    window.setTimeout(function () {
+                        expect(1);
+                        $iframe.remove();
+                        start();
+                    }, 1000);
+                });
+
+                asyncTest('B237815 - Test runner - can\'t execute simple test', function () {
+                    var $iframe        = null;
+                    var callbackRaised = false;
+
+                    var action = function (callback) {
+                        $iframe = $('<iframe id="test_unique_id_kmgvu67">').appendTo('body');
+
+                        window.setTimeout(function () {
+                            $iframe.remove();
+                        }, 50);
+
+                        callback();
+                    };
+
+                    var requestBarrier = new RequestBarrier();
+
+                    action.call(window, function () {
+                        requestBarrier
+                            .wait()
+                            .then(function () {
+                                callbackRaised = true;
+                            });
+                    });
+
+                    window.setTimeout(function () {
+                        ok(callbackRaised);
+                        start();
+                    }, 2000);
+                });
+            });
+    }
+});
```

Comment:
I suppose some of the `fetch` tests are redundant. In fact we haven't changed any logic in `xhr-barrier`, we just add `fetch` event handlers. So I propose leave old tests and add just two for `fetch`: the first with success fetch and the second with failed fetch.